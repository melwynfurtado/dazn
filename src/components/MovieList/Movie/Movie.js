import React from 'react'
import PropTypes from 'prop-types'
import './Movie.scss'

class Movie extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      movieCardOpen: false,
    }
    this.openModal = this.openModal.bind(this)
    this.closeModal = this.closeModal.bind(this)
  }

  openModal(e) {
    this.setState({ movieCardOpen: true })
  }

  closeModal(e) {
    e.stopPropagation()
    this.setState({ movieCardOpen: false })
  }

  render() {
    const { movie } = this.props
    const { movieCardOpen } = this.state

    return (
      <li 
        key={movie.id} 
        className="movie"
        onClick={this.openModal}
      >
        { movie.title }
        { 
          movieCardOpen && 
          <div id="myModal" className="modal">
      
            <div className="modal-content">
              <span className="close" onClick={this.closeModal}>&times;</span>
              <h3>{ movie.title }</h3>
              <p className="clearfix">
                <img 
                  className="poster" 
                  src={ `https://image.tmdb.org/t/p/w200/${movie.poster_path}` } 
                  alt={movie.title} 
                />
                { movie.overview }
              </p>
            </div>
          
          </div>            
        }
      </li>
    )
  }
}

Movie.propTypes = {
  movie: PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    poster_path: PropTypes.string,
    overview: PropTypes.string,
  })
}

export default Movie