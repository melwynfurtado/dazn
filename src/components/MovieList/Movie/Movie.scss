.movie {
  padding: 10px;
  border: 1px solid lightskyblue;
  margin: 3px 0;
  cursor: pointer;

  .poster {
    margin: 0 20px 20px 0;
    float: left;
  }
  
  .clearfix::after {
    content: "";
    clear: both;
    display: table;
  }

  .modal {
    position: fixed;
    z-index: 1;
    left: 0;
    top: 0;
    width: 100%; 
    height: 100%;
    overflow: auto; 
    background-color: rgba(0,0,0,0.05);
  
    .modal-content {
      background-color: #fefefe;
      margin: 15% auto;
      padding: 20px;
      border: 1px solid #888;
      width: 80%;
    }
  
    .close {
      color: #aaa;
      float: right;
      font-size: 28px;
      font-weight: bold;
    }
  
    .close:hover,
    .close:focus {
      color: black;
      text-decoration: none;
      cursor: pointer;
    }    
  }
}