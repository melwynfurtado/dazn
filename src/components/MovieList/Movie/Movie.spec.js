import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Movie from './Movie'

describe('Movie Component', () => {
  let Component

  const movie = {
    id: 123,
    title: 'Rambo II',
    overview: 'Rambo description text'
  }

  beforeEach(() => {
    Component = <Movie movie={movie} />
  })

  it('renders correctly', () => {
    const tree = renderer
                  .create(Component)
                  .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders movie title by default', () => {
    const wrapper = shallow(Component)
    expect(wrapper.find('li.movie').text()).toEqual('Rambo II')
  })

  it('renders movie card on click of movie title', () => {   
    const wrapper = shallow(Component)
    wrapper.find('li.movie').simulate('click')
    expect(wrapper.find('div.modal h3').text()).toEqual('Rambo II')
    expect(wrapper.find('div.modal p').text()).toEqual('Rambo description text')
  })
  
  it('closes movie card on click of close icon on movie card', () => {   
    const wrapper = shallow(Component)
    // Open the modal
    wrapper.find('li.movie').simulate('click')
    // Close the modal
    wrapper.find('li.movie span.close').simulate('click', { stopPropagation : () => {}})
    expect(wrapper.find('div.modal').length).toEqual(0)
  })   
})