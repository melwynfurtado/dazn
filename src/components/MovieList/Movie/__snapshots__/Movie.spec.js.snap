// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Component renders correctly 1`] = `
<li
  className="movie"
  onClick={[Function]}
>
  Rambo II
</li>
`;
